volume-providers/vpc/provider/provider.go

```diff
@@ -62,6 +62,31 @@ func NewProvider(conf *config.Config, logger *zap.Logger) (local.Provider, error
 	if conf.Bluemix == nil || conf.VPC == nil {
 		return nil, errors.New("Incomplete config for VPCBlockProvider")
 	}
+
+	//Do config validation and enable only one generationType (i.e VPC-Classic | VPC-NG)
+	gcConfigFound := (conf.VPC.EndpointURL != "") && (conf.VPC.TokenExchangeURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
+	g2ConfigFound := (conf.VPC.G2EndpointURL != "") && (conf.VPC.G2TokenExchangeURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
+	//if both config found, look for VPCTypeEnabled, otherwise default to GC
+	//Incase of NG configurations, override the base properties.
+	if (gcConfigFound && g2ConfigFound && conf.VPC.VPCTypeEnabled == "g2") || (!gcConfigFound && g2ConfigFound) {
```

Comment:
`conf.VPC.VPCTypeEnabled` seems to be a redundant property, lets not depend on this variable as it needs to be set which even not much useful

Reply:
already replied above..we can discuss, if required.

Reply:
Optional means..it will be empty, when un-set.  VPC provider will set it to "gc" if found EMPTY.
Jenkin now runs NG test with "export VPC_TYPE_ENABLED=g2"